### PR TITLE
bpo-39947: Remove old private trashcan C API functions

### PR DESCRIPTION
**Pull request by vstinner**
Remove 4 C API private trashcan functions which were only kept for
the backward compatibility of the stable ABI with Python 3.8 and
older, since the trashcan API was not usable with the limited C API
on Python 3.8 and older. The trashcan API was excluded from the
limited C API in Python 3.9.

The trashcan C API was never usable with the limited C API, since old
trashcan macros accessed directly PyThreadState members like
"_tstate->trash_delete_nesting", whereas the PyThreadState structure
is opaque in the limited C API.

Exclude also the PyTrash_UNWIND_LEVEL constant from the C API.

The trashcan C API was modified in Python 3.9 by commit
38965ec5411da60d312b59be281f3510d58e0cf1 and in Python 3.10 by commit
ed1a5a5baca8f61e9a99c5be3adc16b1801514fe to hide implementation
details.

https://bugs.python.org/issue39947

### CODE REVIEW

**Comment by vstinner** [comment]
cc @iritkatriel @encukou

File: Misc/NEWS.d/next/C API/2021-06-23-10-31-45.bpo-39947.je_HMo.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Remove 4 private trashcan C API functions which were only kept for the backward
+compatibility of the stable ABI with Python 3.8 and older, since the trashcan
+API was not usable with the limited C API on Python 3.8 and older. The
+trashcan API was excluded from the limited C API in Python 3.9.
```
Comment (encukou) [inline Misc/NEWS.d/next/C API/2021-06-23-10-31-45.bpo-39947.je_HMo.rst:4]: Could you say which ones?

Comment (vstinner) [reply Misc/NEWS.d/next/C API/2021-06-23-10-31-45.bpo-39947.je_HMo.rst:4]: Sure, done.

**Comment by vstinner** [comment]
I updated the PR to list the 4 removed functions in the NEWS entry and I also updated the stable ABI.

**Comment by vstinner** [comment]
@encukou: Would you mind to review the updated PR? It now removes the removed functions from the stable ABI.

**Review by encukou — APPROVED** [review]

**Comment by vstinner** [comment]
Merged, thanks for the review @encukou.

**Pull request merged**